phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -803,7 +766,31 @@ protected PhoenixIndexMetaData getPhoenixIndexMetaData(ObserverContext<RegionCop
         return (PhoenixIndexMetaData)indexMetaData;
     }
 
-    private void preparePostIndexMutations(BatchMutateContext context, long now, PhoenixIndexMetaData indexMetaData)
+    private boolean hasAllIndexedColumns(IndexMaintainer indexMaintainer, MultiMutation multiMutation) {
+        Map<byte[], List<Cell>> familyMap = multiMutation.getFamilyCellMap();
+        for (ColumnReference columnReference : indexMaintainer.getIndexedColumns()) {
+            byte[] family = columnReference.getFamily();
+            List<Cell> cellList = familyMap.get(family);
+            if (cellList == null) {
+                return false;
+            }
+            boolean has = false;
+            for (Cell cell : cellList) {
+                if (Bytes.compareTo(CellUtil.cloneFamily(cell), family) == 0 &&
```

Comment:
Can be replaced with CellUtil.matchingColumn(Cell, family, qualifier)

Reply:
I will use it. Thx